src/shared/ui/components/checkbox/checkbox.types.ts

```diff
@@ -1,4 +1,19 @@
-export interface CheckboxProperties {
-  value: boolean;
+export type CheckboxProperties = {
+  label?: string;
+  disabled?: boolean;
+  className?: string;
+  disabledTooltipText?: string;
+  additionalClassNameWhenChecked?: string;
   onChange: (value: boolean) => void;
-}
+} & (
+  | {
+      value: boolean;
+      type?: 'binary';
+    }
+  | {
+      type: 'extended';
+      value?: ExtendedCheckboxOption;
+    }
+);
```

Comment:
can you explain why do we need that?

Reply:
We are handling the intermediate state of the checkbox, when only a few of the section options are checked:
![image](https://github.com/user-attachments/assets/c5d09b58-4767-4c05-a972-abc00afb3cd5)

Reply:
but right now I think we can just go for true | false | null and get rid of all of that additional code

Reply:
I'll take a look on that

Reply:
I think we should go with `true | false` and this "parent checkbox" could be a special case, maybe normal checkbox could take a prop `indicator: ReactNode` and you can decide to override default indicator (empty or tick icon) with whatever you want.

Something like:
```tsx
const showSomeItemCheckedIndicator = someItemInGroupChecked && !wholeGroupChecked;

// parent checkbox
<Checkbox
  value={wholeGroupChecked}
  onChange={toggleWholeGroupChange}
  indicator={showSomeItemCheckedIndicator ? <DashIcon /> : undefined}
/> 
```

Reply:
thanks, I've refactored the checkbox component as we agreed on Discord